docs/drafts/Documentation Guidelines.md

```diff
@@ -0,0 +1,38 @@
+*NB: This information is still in draft form, and should not be relied upon to inform usage of or contribution to `cubing.js`.*
+
+## Purpose of this document
+
+This document serves to describe `cubing.js`'s approach to documentation in both practical and theoretical terms. Anyone looking to contribute either documentation or new features to `cubing.js` should consult this document.
+
+## Documentation guidelines:
+
+- Write API reference documentation inline using [TSDoc-formatted](https://tsdoc.org/) comments 
+    - PR's lacking these comments will be considered incomplete
```

Comment:
I think this phrasing is a tad aggressive. And if we get a PR that we think needs documentation, a reviewer will almost certainly ask. Could we go for something like "Please include documentation for any new definitions (e.g. classes, functions, types, `const` exports) that someone using the library may interact with."

Reply:
Question here - do you not think documentation should be written even on internal changes? My thinking is that this will make it easier for contributors to onboard, but people contributing to the core codebase may not necessarily be a huge priority, as you and Tomas are getting quite a bit done - and trying to build internal documentation may slow that down for relatively little gain.

Reply:
Also, see updated phrasing